### ISSUE TITLE
Crashing issue when using storage provider

### ISSUE THREAD

**Issue by IceTank** [PrismarineJS/prismarine-world#109]
The saveNow https://github.com/PrismarineJS/prismarine-world/blob/28113873b7a32f52197797374fec2ba08e773a6c/src/world.js#L131 function can call the storageProvider to safe chunks that have already been unloaded. Resulting in this error: 
```
D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-provider-anvil\src\1.8\chunk.js:193
        biomes.push(chunk.getBiome(new Vec3(x, 0, z)))
                          ^

TypeError: Cannot read properties of undefined (reading 'getBiome')
    at writeBiomes (D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-provider-anvil\src\1.8\chunk.js:193:27)
    at prismarineChunkToNbt (D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-provider-anvil\src\1.8\chunk.js:22:21)
    at Anvil.save (D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-provider-anvil\src\anvil.js:44:19)
    at World.saveNow (D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-world\src\world.js:139:30)
    at Timeout._onTimeout (D:\NodeJS\mineflayer-proxy-inspector\node_modules\prismarine-world\src\world.js:147:18)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

My suggesting is to move a reference for the chunk that should be saved into the savingQueue Map. This way every chunk can be saved even if it has already been unloaded. If unloaded chunks are just ignored the saver might skip chunks instead off saving them.

**Comment by kf106** [comment]
I've been using the code from https://github.com/kf106/prismarine-world/blob/master/src/world.js for half a year now, and I've not had any save/load problems or the chunk.getBiome error since then, even when saving 10k chunks. See https://github.com/PrismarineJS/prismarine-world/pull/100

The problem is indeed that the unload is being performed before the save has completed, resulting in junk being saved.

**Comment by IceTank** [comment]
I just noticed that this issue should have been fixed with this pr #100. I don't think I had this issue since.
